modules/sharing-registry/sharing-data-migrator/src/main/java/org/apache/airavata/sharing/registry/migrator/airavata/AiravataDataMigrator.java

```diff
@@ -205,17 +218,56 @@ public static void main(String[] args) throws SQLException, ClassNotFoundExcepti
             }
         }
 
+        //Creating the everyone group
+        List<Domain> domainList = sharingClient.getDomains(-1, 0);
+        for (Domain domain : domainList) {
+            CredentialStoreService.Client credentialStoreServiceClient = getCredentialStoreServiceClient();
+            GatewayResourceProfile gatewayResourceProfile = getRegistryServiceClient().getGatewayResourceProfile(domain.domainId);
+            PasswordCredential credential = credentialStoreServiceClient.getPasswordCredential(
+                    gatewayResourceProfile.getIdentityServerPwdCredToken(), gatewayResourceProfile.getGatewayID());
+            String groupOwner = credential.getLoginUserName();
+
+            String groupId = "everyone@" + domain.domainId;
+            if (!sharingClient.isGroupExists(domain.domainId, groupId)) {
+                UserGroup userGroup = new UserGroup();
+                userGroup.setGroupId(groupId);
+                userGroup.setDomainId(domain.domainId);
+                userGroup.setGroupCardinality(GroupCardinality.MULTI_USER);
+                userGroup.setCreatedTime(System.currentTimeMillis());
+                userGroup.setUpdatedTime(System.currentTimeMillis());
+                userGroup.setName("everyone");
+                userGroup.setDescription("Default Group");
+                userGroup.setOwnerId(groupOwner + "@" + domain.domainId);
+                userGroup.setGroupType(GroupType.DOMAIN_LEVEL_GROUP);
+                sharingClient.createGroup(userGroup);
+
+                List<User> userList = sharingClient.getUsers(domain.domainId, -1, 0);
+                List<String> users = new ArrayList<>();
+                for (User user : userList) {
+                    users.add(user.getUserId());
+                }
+                sharingClient.addUsersToGroup(domain.domainId, users, groupId);
+            }
+        }
+
         //Creating application deployment entries
         query = "SELECT * FROM APPLICATION_DEPLOYMENT";
         statement = expCatConnection.createStatement();
         rs = statement.executeQuery(query);
         while(rs.next()){
             try {
+                CredentialStoreService.Client credentialStoreServiceClient = getCredentialStoreServiceClient();
```

Comment:
You probably don't need this here if you moved the credentialStoreServiceClient out of the loop above.

Reply:
Done.